src/Symfony/Bundle/FrameworkBundle/Secrets/EnvVault.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secrets;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class EnvVault extends SodiumVault
+{
+    private $dotEnvFile;
+
+    public function __construct(string $dotEnvFile)
+    {
+        $this->dotEnvFile = $dotEnvFile;
+    }
+
+    public function generateKeys(bool $override = false): bool
+    {
+        return false;
+    }
+
+    public function seal(string $name, string $value): void
+    {
+        $this->validateName($name);
+        $name .= '_SECRET';
+        $value = rawurlencode($value);
+
+        $content = file_exists($this->dotEnvFile) ? file_get_contents($this->dotEnvFile) : '';
+        $content = preg_replace("/^$name=.*/m", "$name=$value", $content, -1, $count);
+
+        if (!$count) {
+            $content .= "$name=$value\n";
+        }
+
+        file_put_contents($this->dotEnvFile, $content);
+    }
+
+    public function reveal(string $name): ?string
+    {
+        $this->validateName($name);
+        $name .= '_SECRET';
+        $value = \is_string($_SERVER[$name] ?? null) && 0 !== strpos($name, 'HTTP_') ? $_SERVER[$name] : ($_ENV[$name] ?? null);
+
+        return null !== $value ? rawurldecode($value) : null;
+    }
+
+    public function remove(string $name): bool
+    {
+        $this->validateName($name);
+        $name .= '_SECRET';
+
+        $content = file_exists($this->dotEnvFile) ? file_get_contents($this->dotEnvFile) : '';
+        $content = preg_replace("/^$name=.*\n/m", '', $content, -1, $count);
+
+        if ($count) {
+            file_put_contents($this->dotEnvFile, $content);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public function list(bool $reveal = false): array
+    {
+        $secrets = [];
+
+        foreach ($_ENV as $name => $value) {
+            if (preg_match('/^\w+_SECRET$/D', $name)) {
+                $secrets[substr($name, 0, -\strlen('_SECRET'))] = $reveal ? $value : null;
```

Comment:
missing `rawurldecode`

Reply:
there's no reason for that. This is not a URL-based DSN.

Reply:
@stof because the value is urlencoded in the `seal` method